src/v/iceberg/tests/manifest_serialization_test.cc

```diff
@@ -314,6 +315,8 @@ TEST(ManifestSerializationTest, TestMetadataWithPartitionSpec) {
 }
 
 TEST(ManifestSerializationTest, TestSerializeManifestData) {
+    // File may not be on mount that supports O_DIRECT.
+    ss::engine().set_strict_dma(false);
```

Comment:
mmm, why not run your tests on xfs?

Reply:
I suspect runfiles is creating a link to the file rather than copying the file. When I run out of the xfs dir, it results in the same exception:

```
~/Repos/redpanda$ TEST_TMPDIR=~/xfs/.cache bazel test //src/v/iceberg/...
...
C++ exception with description "filesystem error: open failed: Invalid argument ["/home/andrew/xfs/.cache/_bazel_awong/935cfdfb7c7ea48abc7b14ceaec27e80/sandbox/linux-sandbox/3/execroot/_main/bazel-out/k8-fastbuild/bin/src/v/iceberg/tests/manifest_serialization_test.runfiles/_main/src/v/iceberg/tests/testdata/nested_manifest.avro"]" thrown in the test body.
```

Even though the mount is xfs:
```
~/Repos/redpanda$ df --output=fstype /home/awong/xfs
Type
xfs
```